os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -210,42 +202,25 @@ int binary_manager_load_binary(int bin_idx)
 		}
 		bmvdbg("BIN[%d] %s %d %d\n", bin_idx, devname, load_attr.bin_size, load_attr.offset);
 
-		/* If a priority of loaded task is higher than priority of loading thread,
-		 * we need to lock scheduling until updating loading info in binary table. */
-		if (load_attr.priority >= LOADINGTHD_PRIORITY) {
-			sched_lock();
-			is_sched_locked = true;
-		}
-
-		ret = load_binary(devname, &load_attr);
-		if (ret > 0) {
-			bin_pid = (pid_t)ret;
-			bmvdbg("Load '%s' success! pid = %d\n", devname, bin_pid);
-			break;
-		} 
-#if (defined(CONFIG_BUILD_PROTECTED) || defined(CONFIG_BUILD_KERNEL)) && defined(CONFIG_MM_KERNEL_HEAP)
-		else if (errno == ENOMEM) {
-			if (is_sched_locked) {
-				sched_unlock();
-				is_sched_locked = false;
+		retry_count = 0;
```

Comment:
This Logic is NOT right. load binary API failure is NOT the main concern, the main issue is if loaded binary reboots for same reason within an interval it should NOT be re-loaded.

Reply:
I consider what you said more.